File: Codigos estudiantes por lenguaje/Java/Sebastian_Acevedo/Búsquedas/BinarySearch.java
public class Main {

	static int binarySearch(int arr[], int left, int right, int x) {
		if (right>=left) {
			int mid=left+(right-left)/2;
			if(arr[mid]==x) {
				return mid;
			}
			if(arr[mid]>x) {
				return binarySearch(arr, left, mid-1, x);
			}
			if(arr[mid]<x) {
				return binarySearch(arr, mid+1, right, x);
			}
		}
		return -1;
	}
	
    public static void main(String[] args) {
    	int arr[]= {2,3,4,10,40};
    	int n=arr.length;
    	int x=2;
    	int result=binarySearch(arr, 0, n-1, x);
    	if(result==-1)System.out.println("No encontró el dato");
    	else System.out.println("El dato está en la pos "+ result);
    	
    }
}

File: Codigos estudiantes por lenguaje/PY/Acevedo_Sebastian/Búsquedas/binary_Search.py
def binary_Search(arr, left, right, x):
    if right >= left:

        mid = left + (right - left) // 2

        if arr[mid] == x:
            return mid
        elif arr[mid] > x:
            return binary_Search(arr, left, mid-1, x)
        else:
            return binary_Search(arr, mid+1, right, x)
    else:
        return -1


arr = [2, 3, 4, 10, 40, 55, 56, 100]
n = len(arr)
x = int(input())
result = binary_Search(arr, 0, n-1, x)

if result == -1:
    print('Dato no encontrado')
else:
    print(f'Dato en la posicion {result}')
